tools/theme-preview.sh

```diff
@@ -21,10 +21,8 @@ else
   theme=$1
   shift
 
-  for file in "$@"; do
-    # shellcheck disable=SC1090
-    . "$file"
-  done
+  # We will source the given config files after each prompt activation,
+  # so as to allow presets using public functions (like `lp_terminal_format`).
```

Comment:
I don't understand the need for this change. Loading a theme file doesn't run any code (or shouldn't, see the docs:
https://github.com/nojhan/liquidprompt/blame/master/docs/theme/custom.rst#L10-L13
> A theme should be contained in one file with a ``.theme`` file suffix. There should be no "top level" code in the file, or in other words, all code should be contained in functions. Sourcing the file should run no code, as a user sourcing the theme file might not want to activate it yet.

As long as we don't run the theme's activate function until everything is setup, there can't be any issues with loading them this early.

Reply:
For some presets, I need to call `lp_terminal_format`, but it is not available when sourcing presets early on, so that was a way to bypass the limitation. Is there another way of doing that? Having a common interface to set up 8-bit colors is really a need for presets, IMHO.

Reply:
Wait, are you using this script to test config presets? It was never designed to do that, as you have discovered. It was designed for themes.

Hm. I bet your hack here would work. I can't think of any reason that it wouldn't, but if the goal is to support loading both theme and config preset files, the docs on this script should be updated to mention that.

I don't think that you need to source the files multiple times though. Nothing should overwrite the config options later.